chapters/descriptor_buffer.adoc

```diff
@@ -0,0 +1,250 @@
+
+ifndef::chapters[:chapters:]
+ifndef::images[:images: images/]
+
+[[descriptor-buffer]]
+= Descriptor Buffer
+
+This chapter aims to illustrate better how link:https://github.com/KhronosGroup/Vulkan-Docs/blob/main/proposals/VK_EXT_descriptor_buffer.adoc[VK_EXT_descriptor_buffer] mapping of memory works.
+
+The goal here is **not** to show a real example or recommended usage, but instead help understand how the API is mapping data to the shader, so that afterwards you can use this API in any way you want.
+
+[NOTE]
+====
+This will only use Storage Buffers because it's simpler to demonstrate the mappings. Samplers and images work in the same general way, but with caveats better explained in the extension proposal.
+====
+
+== Terminology Overload
+
+To try and clear some terms up first:
+
+* "descriptor buffers" are just `VkBuffer` created with the `VK_BUFFER_USAGE_RESOURCE_DESCRIPTOR_BUFFER_BIT_EXT` flags
+* "samplers" are `VkSampler` (`VK_DESCRIPTOR_TYPE_SAMPLER`)
+* "resources" are all other VkDescriptorType
+** This could be a `VkBuffer`, which can be called a "resource buffer"
+
+== The Example Shader
+
+We will take a basic shader that has 3 sets, each with 3 descriptors inside of them. Each descriptor gets written a unique value.
+
+[source,glsl]
+----
+layout (set = 0, binding = 0) buffer A { uint a; };
+layout (set = 0, binding = 1) buffer B { uint b; };
+layout (set = 0, binding = 2) buffer C { uint c; };
+
+layout (set = 1, binding = 0) buffer D { uint d; };
+layout (set = 1, binding = 1) buffer E { uint e; };
+layout (set = 1, binding = 2) buffer F { uint f; };
+
+layout (set = 2, binding = 0) buffer G { uint g; } array[3];
+
+void main() {
+    a = 10;
+    b = 20;
+    c = 30;
+    d = 40;
+    e = 50;
+    f = 60;
+    array[0].g = 70;
+    array[1].g = 80;
+    array[2].g = 90;
+}
+----
+
+With this shader, we will then have 3 `VkDescriptorSetLayout` in a single pipeline that exactly matches the shader interface.
+
+[source,c++]
+----
+// Set 0 and 1
+VkDescriptorSetLayoutBinding bindings_a[3] {
+    { binding = 0, descriptorType = VK_DESCRIPTOR_TYPE_STORAGE_BUFFER, descriptorCount = 1 },
+    { binding = 1, descriptorType = VK_DESCRIPTOR_TYPE_STORAGE_BUFFER, descriptorCount = 1 },
+    { binding = 2, descriptorType = VK_DESCRIPTOR_TYPE_STORAGE_BUFFER, descriptorCount = 1 }
+}
+
+// Set 2
+VkDescriptorSetLayoutBinding bindings_b[1] {
+    { binding = 0, descriptorType = VK_DESCRIPTOR_TYPE_STORAGE_BUFFER, descriptorCount = 3 }
+}
+
+VkDescriptorSetLayout ds_layout_0(bindings_a); // Set 0
+VkDescriptorSetLayout ds_layout_1(bindings_a); // Set 1
+VkDescriptorSetLayout ds_layout_2(bindings_b); // Set 2
+VkPipelineLayout pipeline_layot([ds_layout_0, ds_layout_1, ds_layout_2]);
+----
+
+== Query Descriptor Set Layout Sizes
+
+Now using the new `vkGetDescriptorSetLayoutSizeEXT` and `vkGetDescriptorSetLayoutBindingOffsetEXT` commands, we can get info from the driver what size it needs to properly use these `VkDescriptorSetLayout`.
```

Comment:
I *generally* try to avoid words like "new" when referencing API calls.  That's a quick way to help ensure that the guide doesn't get dated as soon as those functions are no longer "new."  i.e. if someone were to look up those functions in the spec and see they were introduced in 1.4.x and the current spec is 1.5.x etc they might start to expect that the information is no longer potentially relevant.

Reply:
I guess I meant "new" as in relative to what this extension adds, will re-word it